### PR TITLE
[ML-3585] Added benchmarks to mllib-large.yaml for clustering

### PR DESCRIPTION
**Pull request by ludatabricks**
Benchmark for clustering is added to mllib-large.yaml.
GaussianMixture, KMeans, and LDA are added. BisectingKMeans is missing in spark-sql-perf now. Need to be fixed in the following up JIRA: https://databricks.atlassian.net/browse/ML-3834
Then parameters is based on the previous benchmarks for the Spark 2.2 QA.

### CODE REVIEW

File: src/main/resources/com/databricks/spark/sql/perf/mllib/config/mllib-large.yaml [hunk 1]
```diff
@@ -37,6 +37,28 @@ benchmarks:
...
       smoothing: 1.0
+  - name: clustering.GaussianMixture
+    params:
+      numExamples: 100000
+      numTestExamples: 100000
+      numFeatures: 1000
+      k: 10
+      maxIter: 10
+      tol: 0.01
+  - name: clustering.KMeans
```
Comment (mengxr) [inline src/main/resources/com/databricks/spark/sql/perf/mllib/config/mllib-large.yaml]: what is the dataset size for kmeans?

Comment (ludatabricks) [reply src/main/resources/com/databricks/spark/sql/perf/mllib/config/mllib-large.yaml]: The size is 1000000.

Comment (mengxr) [reply src/main/resources/com/databricks/spark/sql/perf/mllib/config/mllib-large.yaml]: But why it is not a param?

Comment (ludatabricks) [reply src/main/resources/com/databricks/spark/sql/perf/mllib/config/mllib-large.yaml]: We have some common default parameters:
common:
  numExamples: 1000000
  numTestExamples: 1000000
  numFeatures: 4000
  numPartitions: 64
  randomSeed: [1, 1, 1]

These will be the params for the benchmark call, but we don't need to write it for each case.

**Comment by mengxr** [comment]
LGTM

**Pull request merged**